docs/source/metrics.rst

```diff
@@ -1,4 +1,312 @@
-.. automodule:: pytorch_lightning.metrics
-   :members:
-   :noindex:
-   :exclude-members:
+Metrics
+=======
+This is a general package for PyTorch Metrics. These can also be used with regular non-lightning PyTorch code.
+Metrics are used to monitor model performance.
+
+In this package we provide two major pieces of functionality.
+
+1. A Metric class you can use to implement metrics with built-in distributed (ddp) support which are device agnostic.
+2. A collection of popular metrics already implemented for you.
+
+Example:
+
+.. code-block:: python
+
+    from pytorch_lightning.metrics.functional import accuracy
+
+    pred = torch.tensor([0, 1, 2, 3])
+    target = torch.tensor([0, 1, 2, 2])
+
+    # calculates accuracy across all GPUs and all Nodes used in training
+    accuracy(pred, target)
+
+    # tensor(0.7500)
+
+
+--------------
+
+Implement a metric
+------------------
+You can implement metrics as either a PyTorch metric or a Numpy metric. Numpy metrics
+will slow down training, use PyTorch metrics when possible.
+
+Use :class:`TensorMetric` to implement native PyTorch metrics. This class
+handles automated DDP syncing and converts all inputs and outputs to tensors.
+
+Use :class:`NumpyMetric` to implement numpy metrics. This class
+handles automated DDP syncing and converts all inputs and outputs to tensors.
+
+.. warning:: Numpy metrics might slow down your training substantially,
+    since every metric computation requires a GPU sync to convert tensors to numpy.
+
+TensorMetric
+^^^^^^^^^^^^
+Here's an example showing how to implement a TensorMetric
+
+.. code-block:: python
+
+    class RMSE(TensorMetric):
+        def forward(self, x, y):
+            return torch.sqrt(torch.mean(torch.pow(x-y, 2.0)))
+
+.. autoclass:: pytorch_lightning.metrics.metric.TensorMetric
+    :noindex:
+
+NumpyMetric
+^^^^^^^^^^^
+Here's an example showing how to implement a NumpyMetric
+
+.. code-block:: python
+
+    class RMSE(NumpyMetric):
+        def forward(self, x, y):
+            return np.sqrt(np.mean(np.power(x-y, 2.0)))
+        
+
+.. autoclass:: pytorch_lightning.metrics.metric.NumpyMetric
+    :noindex:
+
+--------------
+
+Class Metrics
+-------------
+The following are metrics which can be instantiated as part of a module definition (even with just
+plain PyTorch).
+
+.. code-block:: python
+
+    from pytorch_lightning.metrics import Accuracy
+
+    # Plain PyTorch
+    class MyModule(nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.rmse = RMSE()
```

Comment:
We should probably use accuracy here, since RMSE is not implemented in metrics yet (hope I have time for a PR one of the next days). 
```suggestion
            self.metric = Accuracy()
```